xbmc/cores/AudioEngine/Sinks/AESinkXAudio.cpp

```diff
@@ -354,8 +303,15 @@ void CAESinkXAudio::EnumerateDevicesEx(AEDeviceInfoList &deviceInfoList, bool fo
     deviceInfo.m_channels.Reset();
     deviceInfo.m_dataFormats.clear();
     deviceInfo.m_sampleRates.clear();
+    deviceChannels.Reset();
+
+    for (unsigned int c = 0; c < WASAPI_SPEAKER_COUNT; c++)
+    {
+      if (details.uiChannelMask & WASAPIChannelOrder[c])
+        deviceChannels += AEChannelNames[c];
+    }
 
-    std::wstring deviceId = KODI::PLATFORM::WINDOWS::ToW(details.strDeviceId);
+    std::wstring deviceId = KODI::PLATFORM::WINDOWS::ToW(details.strDeviceRawId);
 
     /* Test format DTS-HD-MA */
```

Comment:
Unless I missed something, XAudio2 only accepts PCM formats. The tests for HD formats passthrough could be removed?

Reply:
Yes, this make sense and also will reduce a bit the devices enumeration time...